openeo/rest/connection.py

```diff
@@ -55,6 +55,21 @@ def __init__(self, http_status_code: int = None,
         self.url = url
         super().__init__("[{s}] {c}: {m}".format(s=self.http_status_code, c=self.code, m=self.message))
 
+def paginate(con, url: str, params: dict = {}, callback: Callable = None):
+    page = 1
+    while True:
+        response = con.get(url, params = params).json()
+        if callback:
+            yield callback(response, page)
+        else:
+            yield response
+        next_link = next((link for link in response['links'] if link['rel'] == 'next'), None)
+        if next_link is None:
+             break
+        url = next_link['href']
+        page += 1
+        params = {}
```

Comment:
Are you intentionally clearing the `params`?  Won't that change the result set , and consequently the paging?

Reply:
Yes, the params must be in the url we get back two lines above.